pandas/_libs/window.pyx

```diff
@@ -443,10 +443,17 @@ def roll_sum(ndarray[double_t] input, int64_t win, int64_t minp,
         double val, prev_x, sum_x = 0
         int64_t s, e
         int64_t nobs = 0, i, j, N
+        int64_t minp2 = -1
         bint is_variable
         ndarray[int64_t] start, end
         ndarray[double_t] output
 
+    if minp == 0:
+        # in get_window_indexer, we ensure that minp >= 1. That's fine for
+        # all cases except nobs = 0 (all missing values) and minp=0. For
```

Comment:
all of the minp logic is handled externally to window.pyx (in window.py)

Reply:
Are you saying to move this there? I don't think that's possible without rewriting `roll_sum`. All of it's logic assumes that `minp` is at least 1. The only thing I need the user-provided `minp` is for passing to `calc_sum` later.

Reply:
then it should be changed. This is way too hacky. ALL logic related to minp is passed in.